packages/hint/src/lib/cli/analyze.ts

```diff
@@ -141,126 +176,117 @@ const askUserToInstallDependencies = async (resources: HintResources): Promise<b
+        const config = Configuration.loadConfigFile(resolvedPath);
 
         return config || getDefaultConfiguration();
     } catch (e) {
-        logger.error(e);
-
+        /*
+         * If there is an error with the configuration,
+         * returns null to ask later to the user.
+         */
         return null;
     }
 };
 
-const messages: { [name: string]: string } = {
-    'fetch::end': '%url% downloaded',
-    'fetch::start': 'Downloading %url%',
-    'scan::end': 'Finishing...',
-    'scan::start': 'Analyzing %url%',
-    'traverse::down': 'Traversing the DOM',
-    'traverse::end': 'Traversing finished',
-    'traverse::start': 'Traversing the DOM',
-    'traverse::up': 'Traversing the DOM'
-};
+const askToInstallPackages = async (resources: HintResources): Promise<boolean> => {
+    if (resources.missing.length > 0) {
+        appInsights.trackEvent('missing', resources.missing);
+    }
 
-const getEvent = (event: string) => {
-    if (event.startsWith('fetch::end')) {
-        return 'fetch::end';
+    if (resources.incompatible.length > 0) {
+        appInsights.trackEvent('incompatible', resources.incompatible);
     }
 
-    return event;
-};
+    const missingPackages = resources.missing.map((name) => {
+        return `@hint/${name}`;
+    });
 
-const setUpUserFeedback = (engine: Engine, spinner: Ora) => {
-    engine.prependAny(((event: string, value: { resource: string }) => {
-        const message: string = messages[getEvent(event)];
+    const incompatiblePackages = resources.incompatible.map((name) => {
+        // If the packages are incompatible, we need to force to install the latest version.
+        return `@hint/${name}@latest`;
+    });
 
-        if (!message) {
-            return;
-        }
+    if (!(await askUserToInstallDependencies(resources) &&
+        await installPackages(missingPackages) &&
+        await installPackages(incompatiblePackages))) {
 
-        spinner.text = message.replace('%url%', cutString(value.resource));
-    }) as EventAndListener);
+        // The user doesn't want to install the dependencies or something went wrong installing them
+        return false;
+    }
+
+    // After installing all the packages, we need to load the resources again.
+    return true;
 };
 
-/** Asks the users if they want to create a new configuration file or use the default one. */
-const getDefaultOrCreateConfig = async (actions: CLIOptions): Promise<Configuration | null> => {
-    const useDefault = await askUserToUseDefaultConfiguration();
-    let userConfig: UserConfig;
+const getAnalyzer = async (userConfig: UserConfig, options: CreateAnalyzerOptions): Promise<Analyzer> => {
+    let webhint: Analyzer;
 
-    if (useDefault) {
-        userConfig = getDefaultConfiguration();
-    } else {
-        logger.error(`Unable to find a valid configuration file. Please create a valid .hintrc file using 'npm init hintrc'.`);
+    try {
+        webhint = Analyzer.create(userConfig, options);
```

Comment:
Even though this is a static method, we should still create a `createAnalyzer` wrapper that we export to keep things simple and consume that both here and in the vscode extension.

```suggestion
        webhint = createAnalyzer(userConfig, options);
```